specs/separate_configuration_files_per_component.adoc

```diff
@@ -0,0 +1,246 @@
+= Specification for segregating configurations per module wise
+
+This specification describes, Currently all the configuration details of
+different modules are read from same file. Instead of that store the
+configuration details in different files as well as etcd and read from
+particular source based on priority.
+
+== Problem description
+
+Currently, All the tendrl modules are configured from same file. Also, while
+import flow the configuration file gets appended with SDS specific
+configurations while installation of SDS integration bits on the storage nodes.
+To avoid this we have to store the configuration details into layer wise.
+
+1. Store the configuration details into different files as per modules, And
+   configure modules using those files. (high priority).
+2. Each modules should store its own configuration details to etcd and configured
+   via api request (Less priority).
+
+At the time of installation, Each module's configuration details should store in
+different files with proper name. Always tendrl modules should configured from
+those files only.
+When module run the configuration details should be stored in etcd except
+common library.
+
+* Integration modules should store its configuration details under cluster
+  specific hierarchy in the central store.
+* Other modules should store its configuration with proper hierarchy.
+
+The configuration in the central store needs to be treated as an object
+hierarchy, which appropriate atoms and flows to allow modification of this
+object hierarchy. If any modification on configurations in etcd, Then it should
+affect a local configuration files also. But any changes in local configuration
+file should not affect configurations in etcd. Any changes in integration
+specific configuration in etcd will affect config files of particular cluster
+nodes only. But changes in other configuration in etc will affect all nodes
+local config files.
+
+== Use Cases
+
+Centralized deployment-wide or cluster-wide configuration in the central store,
+which can be overridden per component via configuration files or command line
+options.
+
+== Proposed change
+
+* Modify the sample configuration files in each modules. It should contain
+  its own and relevant configuration information only.
+* Rename the sample configuration files in each modules with proper name except
```

Comment:
Common library should not have conf sample and the same would be taken care as part of re-factoring task. The module which uses common module, would impose its configurations to common module while logging etc.